plugins/outputs/websocket/websocket.go

```diff
@@ -0,0 +1,206 @@
+package websocket
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+
+	ws "github.com/gorilla/websocket"
+)
+
+var sampleConfig = `
+  ## URL is the address to send metrics to. Make sure ws or wss scheme is used.
+  url = "ws://127.0.0.1:8080/telegraf"
+
+  ## Timeouts (make sure read_timeout is larger than server ping interval or set to zero).
+  # connect_timeout = "5s"
+  # write_timeout = "5s"
+  # read_timeout = "30s"
+
+  ## Optionally turn on using text data frames (binary by default).
+  # use_text_frames = false
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  # data_format = "influx"
+
+  ## Additional HTTP Upgrade headers
+  # [outputs.websocket.headers]
+  #   Authorization = "Bearer <TOKEN>"
+`
+
+const (
+	defaultConnectTimeout = 5 * time.Second
+	defaultWriteTimeout   = 5 * time.Second
```

Comment:
usually like to see defaults closer to 30s or 60s.

Reply:
Made defaults 10 seconds for connect and write. Not sure that 30 or even 60 seconds make lots of sense as default values since this seems too large. Is it ok or you want me to increase those more?

Reply:
large writes on slow networks, or networks with temporary drops could take longer than 10s. I've seen 6s writes on a local machine to docker. The goal is not to drop a connection that is working, but slow.

Reply:
OK, got it, done - 30s for all